src/compare-internal.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <vips/vips.h>
+
+#include "composite.h"
+
+const int STATS_SUM2_COLUMN = 3;
+
+/*
+  Compare images `actual` and `expected` and return mean squared error (MSE).
+ */
+int Compare(VipsObject *context, VipsImage *actual, VipsImage *expected, double *out) {
+  if (actual->Bands != expected->Bands)
+    return -1;
+
+  if (actual->Type != expected->Type)
+    return -1;
+
+  VipsImage *actualPremultiplied;
+  VipsImage *expectedPremultiplied;
+  if (Premultiply(context, actual, &actualPremultiplied) ||
```

Comment:
Does the use of `Premultiply` here need wrapping in the `shouldPremultiplyAlpha` logic?